integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/mock/UsedImagesTest.java

```diff
@@ -0,0 +1,180 @@
+
+    @Inject
+    private ImageCatalogTestClient imageCatalogTestClient;
+
+    @Inject
+    private SdxTestClient sdxTestClient;
+
+    @Inject
+    private UtilTestClient utilTestClient;
+
+    @Inject
+    private FreeIpaTestClient freeIpaTestClient;
+
+    @Inject
+    private CloudbreakActor cloudbreakActor;
+
+    @Override
+    protected void setupTest(TestContext testContext) {
+        createDefaultUser(testContext);
+        createDefaultCredential(testContext);
+        createDefaultImageCatalog(testContext);
+        initializeDefaultBlueprints(testContext);
+    }
+
+    @Test(dataProvider = TEST_CONTEXT_WITH_MOCK)
+    @Description(
+            given = "a running environment with an sdx",
+            when = "list used images requests are sent to cb and freeipa",
+            then = "response should have the images"
+    )
+    public void testUsedImages(MockedTestContext testContext) {
+        String cbImageCatalogName = resourcePropertyProvider().getName();
+        CloudbreakUser internalActor = cloudbreakActor.createInternal(testContext.getActingUserCrn().getAccountId());
+
+        testContext
+                .given(ImageCatalogTestDto.class)
+                    .withName(cbImageCatalogName)
+                    .withUrl(getImageCatalogMockServerSetup().getPreWarmedImageCatalogUrl())
+                .when(imageCatalogTestClient.createV4())
+
+                .given(EnvironmentNetworkTestDto.class)
+                .given(EnvironmentTestDto.class)
+                    .withNetwork()
+                    .withCreateFreeIpa(Boolean.FALSE)
+                .when(environmentTestClient.create())
+                .await(EnvironmentStatus.AVAILABLE)
+                .when(environmentTestClient.describe())
+
+                .given(FreeIpaTestDto.class)
+                    .withCatalog(getImageCatalogMockServerSetup().getFreeIpaImageCatalogUrl())
+                .when(freeIpaTestClient.create())
+                .await(Status.AVAILABLE)
+                .when(freeIpaTestClient.describe())
+
+                .given(SdxInternalTestDto.class)
+                    .withImageCatalogNameAndImageId(cbImageCatalogName, SDX_IMAGE_UUID)
+                .when(sdxTestClient.createInternal())
+                .await(SdxClusterStatusResponse.RUNNING)
+                .when(sdxTestClient.describeInternal())
+
+                .then((tc, testDto, client) -> verifyImagesAreUsed(tc, testDto, internalActor))
+                .validate();
+
+        testContext
+                .as(cloudbreakActor.defaultUser())
+
+                .given(EnvironmentTestDto.class)
+                .when(environmentTestClient.delete())
+                .await(EnvironmentStatus.ARCHIVED)
+
+                .then((tc, testDto, client) -> verifySdxImageIsNotUsed(tc, testDto, internalActor))
+                .validate();
+    }
+
+    private <T extends CloudbreakTestDto> T verifyImagesAreUsed(TestContext testContext, T testDto, CloudbreakUser internalActor) {
```

Comment:
Some special validation would be useful here for the case: `this can be called only with internal actor only.`. So here would introduce some `whenException` step, where the authorisation error is expected for describe action with non internal user.

Reply:
Great idea, I will add it, thanks!